crates/ironrdp-graphics/src/pointer.rs

```diff
@@ -0,0 +1,379 @@
+use ironrdp_pdu::{
+    cursor::ReadCursor,
+    pointer::{ColorPointerAttribute, LargePointerAttribute, PointerAttribute},
+};
+use thiserror::Error;
+
+/// References:
+/// Drawing pointers: https://learn.microsoft.com/en-us/windows-hardware/drivers/display/pointer-drawing
+/// Drawing color pointers: https://learn.microsoft.com/en-us/windows-hardware/drivers/display/drawing-color-pointers
+/// Drawing monochrome pointers https://learn.microsoft.com/en-us/windows-hardware/drivers/display/drawing-monochrome-pointers
+///
+///
+/// andMask is always 1bpp.
+/// andMask == 0 -> Copy pixel from xorMask
+/// andMask == 1, xorMask == 0 -> Transparent pixel
+/// andMask == 1, xorMask == 1 -> Pixel is inverted
+///
+/// Input: andMask, xorMask, lastFrame -> Output: pixelColor, invertionMask
+
+#[derive(Debug, Error)]
+pub enum PointerError {
+    #[error("Invalid pointer xorMask size. Expected: {expected}, actual: {actual}")]
+    InvalidXorMaskSize { expected: usize, actual: usize },
+    #[error("Invalid pointer andMask size. Expected: {expected}, actual: {actual}")]
+    InvalidAndMaskSize { expected: usize, actual: usize },
+    #[error("Not supported pointer bpp: {bpp}")]
+    NotSupportedBpp { bpp: usize },
+    #[error(transparent)]
+    Pdu(#[from] ironrdp_pdu::PduError),
+}
```

Comment:
Nit: note that at some point I want to remove the dependency on `thiserror`  from all the core-tier crates. The idea is to not depend on heavy proc macros for these so they can be compiled faster and earlier during the build (prior to `syn`). You don’t need to refactor now though.